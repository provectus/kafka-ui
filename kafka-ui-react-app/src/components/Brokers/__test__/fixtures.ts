import { BrokersLogdirs } from 'generated-sources';

export const brokersPayload = [
  { id: 1, host: 'b-1.test.kafka.amazonaws.com', port: 9092 },
  { id: 2, host: 'b-2.test.kafka.amazonaws.com', port: 9092 },
];

export const clusterStatsPayload = {
  brokerCount: 2,
  activeControllers: 1,
  onlinePartitionCount: 138,
  offlinePartitionCount: 0,
  inSyncReplicasCount: 239,
  outOfSyncReplicasCount: 0,
  underReplicatedPartitionCount: 0,
  diskUsage: [
    { brokerId: 0, segmentSize: 334567, segmentCount: 245 },
    { brokerId: 1, segmentSize: 12345678, segmentCount: 121 },
  ],
  version: '2.2.1',
};

const partition = {
  broker: 2,
  offsetLag: 0,
  partition: 2,
  size: 0,
};
const topics = {
  name: '_confluent-ksql-devquery_CTAS_NUMBER_OF_TESTS_59-Aggregate-Aggregate-Materialize-changelog',
  partitions: [partition],
};

export const brokerLogDirsPayload: BrokersLogdirs[] = [
  {
    error: 'NONE',
    name: '/opt/kafka/data-0/logs',
    topics: [
      {
        ...topics,
        partitions: [partition, partition, partition],
      },
      topics,
      {
        ...topics,
        partitions: [],
      },
    ],
  },
];
