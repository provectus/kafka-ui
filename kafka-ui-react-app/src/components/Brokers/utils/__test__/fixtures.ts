import { BrokerMetrics } from 'generated-sources';

export const brokerMetricsPayload: BrokerMetrics = {
  segmentSize: 23,
  segmentCount: 23,
  metrics: [
    {
      name: 'TotalFetchRequestsPerSec',
      labels: {
        canonicalName:
          'kafka.server:name=TotalFetchRequestsPerSec,topic=_connect_status,type=BrokerTopicMetrics',
      },
      value: 10,
    },
    {
      name: 'ZooKeeperRequestLatencyMs',
      value: 11,
    },
    {
      name: 'RequestHandlerAvgIdlePercent',
    },
  ],
};
export const transformedBrokerMetricsPayload =
  '{"segmentSize":23,"segmentCount":23,"metrics":[{"name":"TotalFetchRequestsPerSec","labels":{"canonicalName":"kafka.server:name=TotalFetchRequestsPerSec,topic=_connect_status,type=BrokerTopicMetrics"},"value":10},{"name":"ZooKeeperRequestLatencyMs","value":11},{"name":"RequestHandlerAvgIdlePercent"}]}';
