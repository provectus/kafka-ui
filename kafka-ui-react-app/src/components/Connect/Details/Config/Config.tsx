import React from 'react';
import useAppParams from 'lib/hooks/useAppParams';
import { Controller, useForm } from 'react-hook-form';
import { ErrorMessage } from '@hookform/error-message';
import { yupResolver } from '@hookform/resolvers/yup';
import { RouterParamsClusterConnectConnector } from 'lib/paths';
import yup from 'lib/yupExtended';
import Editor from 'components/common/Editor/Editor';
import { Button } from 'components/common/Button/Button';
import {
  useConnectorConfig,
  useUpdateConnectorConfig,
} from 'lib/hooks/api/kafkaConnect';

import {
  ConnectEditWarningMessageStyled,
  ConnectEditWrapperStyled,
} from './Config.styled';

const validationSchema = yup.object().shape({
  config: yup.string().required().isJsonObject(),
});

interface FormValues {
  config: string;
}

const Config: React.FC = () => {
  const routerParams = useAppParams<RouterParamsClusterConnectConnector>();
  const { data: config } = useConnectorConfig(routerParams);
  const mutation = useUpdateConnectorConfig(routerParams);

  const {
    handleSubmit,
    control,
    reset,
    formState: { isDirty, isSubmitting, isValid, errors },
    setValue,
  } = useForm<FormValues>({
    mode: 'onTouched',
    resolver: yupResolver(validationSchema),
    defaultValues: {
      config: JSON.stringify(config, null, '\t'),
    },
  });

  React.useEffect(() => {
    if (config) {
      setValue('config', JSON.stringify(config, null, '\t'));
    }
  }, [config, setValue]);

  const onSubmit = async (values: FormValues) => {
    try {
      const requestBody = JSON.parse(values.config.trim());
      await mutation.mutateAsync(requestBody);
      reset(values);
    } catch (e) {
      // do nothing
    }
  };

  const hasCredentials = JSON.stringify(config, null, '\t').includes(
    '"******"'
  );
  return (
    <ConnectEditWrapperStyled>
      {hasCredentials && (
        <ConnectEditWarningMessageStyled>
          Please replace ****** with the real credential values to avoid
          accidentally breaking your connector config!
        </ConnectEditWarningMessageStyled>
      )}
      <form onSubmit={handleSubmit(onSubmit)} aria-label="Edit connect form">
        <div>
          <Controller
            control={control}
            name="config"
            render={({ field }) => (
              <Editor {...field} readOnly={isSubmitting} />
            )}
          />
        </div>
        <div>
          <ErrorMessage errors={errors} name="config" />
        </div>
        <Button
          buttonSize="M"
          buttonType="primary"
          type="submit"
          disabled={!isValid || isSubmitting || !isDirty}
        >
          Submit
        </Button>
      </form>
    </ConnectEditWrapperStyled>
  );
};

export default Config;
