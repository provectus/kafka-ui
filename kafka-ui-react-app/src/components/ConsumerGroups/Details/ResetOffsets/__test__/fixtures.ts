export const expectedOutputs = {
  EARLIEST: {
    partitions: [0],
    partitionsOffsets: [
      {
        offset: undefined,
        partition: 0,
      },
    ],
    resetType: 'EARLIEST',
    topic: '__amazon_msk_canary',
  },
  LATEST: {
    partitions: [0],
    partitionsOffsets: [
      {
        offset: undefined,
        partition: 0,
      },
    ],
    resetType: 'LATEST',
    topic: '__amazon_msk_canary',
  },
  OFFSET: {
    partitions: [0],
    partitionsOffsets: [
      {
        offset: '10',
        partition: 0,
      },
    ],
    resetType: 'OFFSET',
    topic: '__amazon_msk_canary',
  },
};
