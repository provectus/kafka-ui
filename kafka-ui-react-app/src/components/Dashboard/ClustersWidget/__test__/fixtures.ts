import { Cluster, ServerStatus } from 'generated-sources';

export const onlineCluster: Cluster = {
  name: 'secondLocal',
  defaultCluster: false,
  status: ServerStatus.Online,
  brokerCount: 1,
  onlinePartitionCount: 6,
  topicCount: 3,
  bytesInPerSec: 0.000030618196853764715,
  bytesOutPerSec: 5.737800890036267075817,
};

export const offlineCluster: Cluster = {
  name: 'local',
  defaultCluster: true,
  status: ServerStatus.Offline,
  brokerCount: 1,
  onlinePartitionCount: 2,
  topicCount: 2,
  bytesInPerSec: 8000.0000067376808542600021,
  bytesOutPerSec: 0.8153063567297119490871,
};

export const clusters: Cluster[] = [onlineCluster, offlineCluster];
