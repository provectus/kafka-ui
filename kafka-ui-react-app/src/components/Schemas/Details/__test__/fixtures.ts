import { SchemaSubject, SchemaType } from 'generated-sources';
import {
  schemaVersion1,
  schemaVersion2,
  schemaVersionWithNonAsciiChars,
} from 'redux/reducers/schemas/__test__/fixtures';

export const versionPayload = [
  schemaVersion1,
  schemaVersion2,
  schemaVersionWithNonAsciiChars,
];
export const versionEmptyPayload = [];

export const jsonSchema: SchemaSubject = {
  subject: 'test',
  version: '15',
  id: 1,
  schema:
    '{"type":"record","name":"MyRecord1","namespace":"com.mycompany","fields":[{"name":"id","type":"long"}]}',
  compatibilityLevel: 'BACKWARD',
  schemaType: SchemaType.JSON,
};

export const protoSchema: SchemaSubject = {
  subject: 'test_proto',
  version: '1',
  id: 2,
  schema:
    'syntax = "proto3";\npackage com.indeed;\n\nmessage MyRecord {\n  int32 id = 1;\n  string name = 2;\n}\n',
  compatibilityLevel: 'BACKWARD',
  schemaType: SchemaType.PROTOBUF,
};
