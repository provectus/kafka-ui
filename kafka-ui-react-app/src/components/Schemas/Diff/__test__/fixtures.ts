import { SchemaSubject, SchemaType } from 'generated-sources';

export const versions: SchemaSubject[] = [
  {
    subject: 'test',
    version: '3',
    id: 3,
    schema:
      'syntax = "proto3";\npackage com.indeed;\n\nmessage MyRecord {\n  int32 id = 1;\n  string name = 2;\n}\n',
    compatibilityLevel: 'BACKWARD',
    schemaType: SchemaType.PROTOBUF,
  },
  {
    subject: 'test',
    version: '2',
    id: 2,
    schema:
      '{"type":"record","name":"MyRecord2","namespace":"com.mycompany","fields":[{"name":"id","type":"long"}]}',
    compatibilityLevel: 'BACKWARD',
    schemaType: SchemaType.JSON,
  },
  {
    subject: 'test',
    version: '1',
    id: 1,
    schema:
      '{"type":"record","name":"MyRecord1","namespace":"com.mycompany","fields":[{"name":"id","type":"long"}]}',
    compatibilityLevel: 'BACKWARD',
    schemaType: SchemaType.JSON,
  },
];
