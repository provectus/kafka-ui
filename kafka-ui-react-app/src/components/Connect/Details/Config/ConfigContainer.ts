import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';
import { RootState } from 'redux/interfaces';
import { fetchConnectorConfig } from 'redux/reducers/connect/connectSlice';
import {
  getIsConnectorConfigFetching,
  getConnectorConfig,
} from 'redux/reducers/connect/selectors';

import Config from './Config';

const mapStateToProps = (state: RootState) => ({
  isConfigFetching: getIsConnectorConfigFetching(state),
  config: getConnectorConfig(state),
});

const mapDispatchToProps = {
  fetchConfig: fetchConnectorConfig,
};

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(Config));
