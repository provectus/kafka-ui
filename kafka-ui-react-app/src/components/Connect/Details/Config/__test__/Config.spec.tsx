import React from 'react';
import { render, WithRoute } from 'lib/testHelpers';
import { clusterConnectConnectorConfigPath } from 'lib/paths';
import Config from 'components/Connect/Details/Config/Config';
import { screen } from '@testing-library/dom';
import { useConnectorConfig } from 'lib/hooks/api/kafkaConnect';
import { connector } from 'lib/fixtures/kafkaConnect';

jest.mock('components/common/Editor/Editor', () => () => (
  <div>mock-Editor</div>
));
jest.mock('lib/hooks/api/kafkaConnect', () => ({
  useConnectorConfig: jest.fn(),
}));

describe('Config', () => {
  const renderComponent = () =>
    render(
      <WithRoute path={clusterConnectConnectorConfigPath()}>
        <Config />
      </WithRoute>,
      {
        initialEntries: [
          clusterConnectConnectorConfigPath(
            'my-cluster',
            'my-connect',
            'my-connector'
          ),
        ],
      }
    );

  it('is empty when no config', () => {
    (useConnectorConfig as jest.Mock).mockImplementation(() => ({}));
    renderComponent();
    expect(screen.queryByText('mock-Editor')).not.toBeInTheDocument();
  });

  it('renders editor', () => {
    (useConnectorConfig as jest.Mock).mockImplementation(() => ({
      data: connector.config,
    }));
    renderComponent();
    expect(screen.getByText('mock-Editor')).toBeInTheDocument();
  });
});
