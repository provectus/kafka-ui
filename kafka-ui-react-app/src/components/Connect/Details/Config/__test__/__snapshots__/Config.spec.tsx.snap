// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Config view matches snapshot 1`] = `
<div
  className="sc-bdfBQB dvPywA"
>
  <mock-JSONEditor
    highlightActiveLine={false}
    isFixedHeight={true}
    readOnly={true}
    value="{
	\\"connector.class\\": \\"FileStreamSource\\",
	\\"tasks.max\\": \\"10\\",
	\\"topic\\": \\"test-topic\\",
	\\"file\\": \\"/some/file\\"
}"
  />
</div>
`;

exports[`Config view matches snapshot when fetching config 1`] = `<mock-PageLoader />`;
